odl/contrib/fom/fom.py

```diff
@@ -0,0 +1,434 @@
+
+    Notes
+    ----------
+    The FOM evaluates
+
+    .. math::
+        1 - \\frac{\| f - g \|^2_2}{\| f \|^2_2 + \| g \|^2_2}.
+
+    The FOM takes its values in [0 , 1], with higher correspondance
+    at higher FOM value.
+
+    Parameters
+    ----------
+    test_data : `LinearSpace`
+        Input data or reconstruction.
+    ref_data : `LinearSpace`
+        Reference data to compare 'reco' to.
+    mask : `LinearSpace`
+        Binary mask to define ROI in which FOM evaluation is performed.
+
+    Returns
+    -------
+    fom : scalar
+        Scalar (float) indicating perfect correspondance (1) to no
+        correspondance (0)
+    """
+
+    import odl
+
+    l2_normSquared = odl.solvers.L2NormSquared(test_data.space)
+
+    if mask:
+        test_data = test_data * mask
+        ref_data = ref_data * mask
+
+    diff = test_data - ref_data
+    fom = 1.0 - l2_normSquared(diff) / (l2_normSquared(test_data) +
+                                        l2_normSquared(ref_data))
+    return fom
+
+
+def mean_absolute_error(test_data, ref_data, mask=None):
+    """Mean absolute error-based FOM between test_data and ref_data.
+
+    Evaluates mean absolute error FOM between input (test_data) and
+    reference data (ref_data), by measuring consistency using the L1-norm.
+    Allows for a masking of the two spaces (mask).
+
+    Notes
+    ----------
+    The FOM evaluates
+
+    .. math::
+        1 - \\frac{\| f - g \|_1}{\| f \|_1 + \| g \|_1}.
+
+    The FOM takes its values in [0 , 1], with higher correspondance
+    at higher FOM value.
+
+    Parameters
+    ----------
+    test_data : `LinearSpace`
+        Input data or reconstruction.
+    ref_data : `LinearSpace`
+        Reference data to compare 'reco' to.
+    mask : `LinearSpace`
+        Binary mask to define ROI in which FOM evaluatoin is performed.
+
+    Returns
+    -------
+    fom: scalar
+        Scalar (float) indicating perfect correspondance (1) to no
+        correspondance (0)
+    """
+
+    import odl
+
+    l1_norm = odl.solvers.L1Norm(test_data.space)
+    if mask:
+        test_data = test_data * mask
+        ref_data = ref_data * mask
+    diff = test_data - ref_data
+    fom = 1.0 - l1_norm(diff) / (l1_norm(test_data) + l1_norm(ref_data))
+    return fom
+
+
+def mean_density_value(test_data, ref_data, mask=None):
+    """Mean density value-based FOM betweent test_data and ref_data.
+
+    Evaluates mean density value FOM between input (test_data) and
+    reference data (ref_data). Allows for a masking of the two spaces (mask).
+
+    Notes
+    ----------
+    The FOM evaluates
+
+    .. math::
+        1 - \\bigg \\lvert \\frac{ \\overline{f} - \\overline{g} }
+                                 {\\overline{f} + \\overline{g}} \\bigg
+                                                                 \\rvert.
+
+    where
+
+    .. math::
+        \\overline{f} := \\frac{1}{\|1_\Omega\|_1} \\int_\Omega f dx,
```

Comment:
I guess the `\\frac{1}{\|1_\Omega\|_1}` part will cancel as long as we use normalization?